.circleci/config.yml

```diff
@@ -1,45 +1,230 @@
+          key: v1-bundler-rails-{{ checksum "Gemfile.lock" }}-<< parameters.ruby >>
 
+jobs:
+  unit:
+    parallelism: 1
+    working_directory: ~/knapsack_pro-ruby
+    docker:
+      - image: cimg/ruby:3.2
+    steps:
+      - setup_knapsack_pro_ruby
+      - run: gem install rubocop
+      - run: rubocop --fail-level A --only Style/FrozenStringLiteralComment,Layout/EmptyLineAfterMagicComment lib/
       - run: bundle exec rspec spec
-
       - run: bundle exec ruby spec/knapsack_pro/formatters/time_tracker_specs.rb
+
+  integration-regular:
+    parallelism: 4
+    working_directory: ~/knapsack_pro-ruby
+    parameters:
+      ruby:
+        type: string
+      rspec:
+        type: string
+    docker:
+      - image: cimg/ruby:<< parameters.ruby >>-browsers
+        environment:
+          PGHOST: 127.0.0.1
+          PGUSER: rails-app-with-knapsack_pro
+          RAILS_ENV: test
+          RACK_ENV: test
+          KNAPSACK_PRO_ENDPOINT: https://api-staging.knapsackpro.com
+          KNAPSACK_PRO_TEST_SUITE_TOKEN_RSPEC: 3fa64859337f6e56409d49f865d13fd7
+      - image: cimg/postgres:16.1
+        environment:
+          POSTGRES_DB: rails-app-with-knapsack_pro_test
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: rails-app-with-knapsack_pro
+    steps:
+      - setup_knapsack_pro_ruby
+      - setup_rails_app_with_knapsack_pro:
+          path: ~/rails-app-with-knapsack_pro
+          ruby: << parameters.ruby >>
+          rspec: << parameters.rspec >>
+      - run: dockerize -wait tcp://localhost:5432 -timeout 1m
+      - run:
+          working_directory: ~/rails-app-with-knapsack_pro
+          command: |
+            ruby --version
+            bundle exec rspec --version
+            RSPEC=$(bundle exec rspec --version | grep rspec-core | head -n1 | cut -d " " -f5)
+            [ $RSPEC != << parameters.rspec >> ] && exit 1 || echo "Correct version of RSpec installed: $RSPEC"
+      - run:
+          working_directory: ~/rails-app-with-knapsack_pro
+          command: bin/rails db:setup
+      - run:
+          working_directory: ~/rails-app-with-knapsack_pro
+          command: |
+            export KNAPSACK_PRO_BRANCH="$CIRCLE_BRANCH--$CIRCLE_BUILD_NUM--regular"
+            bundle exec rake knapsack_pro:rspec
+      - run:
+          working_directory: ~/rails-app-with-knapsack_pro
+          command: |
+            # fallback ||
+            export KNAPSACK_PRO_ENDPOINT=https://api-fake.knapsackpro.com
+            export KNAPSACK_PRO_MAX_REQUEST_RETRIES=1
+            bundle exec rake knapsack_pro:rspec
+      - run:
+          working_directory: ~/rails-app-with-knapsack_pro
+          command: |
+            # split by test examples ||
+            export KNAPSACK_PRO_BRANCH="$CIRCLE_BRANCH--$CIRCLE_BUILD_NUM--regular--split"
+            export KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES=true
+            export KNAPSACK_PRO_SLOW_TEST_FILE_PATTERN="{spec/timecop_spec.rb}"
```

Comment:
```suggestion
```

I would add `EXTRA_TEST_FILES_DELAY=10` to global env vars for the docker image.

It would slow down the [`spec/slow_shared_examples_spec.rb`](https://github.com/KnapsackPro/rails-app-with-knapsack_pro/blob/675a1188f39cf8a67ffb447a4487db50fcf98619/spec/slow_shared_examples_spec.rb#L4), which we expect to be split by test examples.

Also I noticed the `spec/foo_spec.rb` is already slow 10s for [the build](https://staging.knapsackpro.com/dashboard/organizations/1/projects/55/test_suites/124/builds/fb7d0c29-aa8f-4b3c-a958-cf567ce1ab37#tab-test-files). So it could be potentially automatically split by examples.

This way we would be closer to what people use as a default. Using `KNAPSACK_PRO_SLOW_TEST_FILE_PATTERN` is not a default approach.
When you use `KNAPSACK_PRO_SLOW_TEST_FILE_PATTERN` then there is no request made to the build_distributions#last API to get the list of tests.

--- 

Or we could leave it as it is if you prefer that. At least we are certain that 1 test file is split by examples (no implicit assumption that we have some slow test file that is supposed to be split).

---

Using `KNAPSACK_PRO_SLOW_TEST_FILE_PATTERN` can be helpful if we would expect a specific path to be split by examples and we test this by checking API response for recorded build subsets (but that's not the scope of this story).